File: e2e/java/src/test/java/dev/kreuzberg/e2e/AuthTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import dev.kreuzberg.kreuzcrawl.Kreuzcrawl;

/** E2e tests for category: auth. */
class AuthTest {
    @Test
    void testAuthBasicHttp() throws Exception {
        // Sends HTTP Basic authentication header
        var engine = Kreuzcrawl.createEngine(null);
        String url = System.getenv("MOCK_SERVER_URL") + "/fixtures/auth_basic_http";
        var result = Kreuzcrawl.scrape(engine, url);
        assertEquals(true, result.authHeaderSent());
        assertEquals(200, result.statusCode());
    }

    @Test
    void testAuthBearerToken() throws Exception {
        // Sends Bearer token in Authorization header
        var engine = Kreuzcrawl.createEngine(null);
        String url = System.getenv("MOCK_SERVER_URL") + "/fixtures/auth_bearer_token";
        var result = Kreuzcrawl.scrape(engine, url);
        assertEquals(true, result.authHeaderSent());
        assertEquals(200, result.statusCode());
    }

    @Test
    void testAuthCustomHeader() throws Exception {
        // Sends authentication via custom header (X-API-Key)
        var engine = Kreuzcrawl.createEngine(null);
        String url = System.getenv("MOCK_SERVER_URL") + "/fixtures/auth_custom_header";
        var result = Kreuzcrawl.scrape(engine, url);
        assertEquals(true, result.authHeaderSent());
        assertEquals(200, result.statusCode());
    }

}

File: e2e/go/auth_test.go
// E2e tests for category: auth
package e2e_test

import (
	"os"
	"testing"

	pkg "github.com/kreuzberg-dev/kreuzcrawl"
)

func Test_AuthBasicHttp(t *testing.T) {
	// Sends HTTP Basic authentication header
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/auth_basic_http"
	result, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	if result.AuthHeaderSent != true {
		t.Errorf("equals mismatch: got %v", result.AuthHeaderSent)
	}
	if result.StatusCode != 200 {
		t.Errorf("equals mismatch: got %v", result.StatusCode)
	}
}

func Test_AuthBearerToken(t *testing.T) {
	// Sends Bearer token in Authorization header
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/auth_bearer_token"
	result, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	if result.AuthHeaderSent != true {
		t.Errorf("equals mismatch: got %v", result.AuthHeaderSent)
	}
	if result.StatusCode != 200 {
		t.Errorf("equals mismatch: got %v", result.StatusCode)
	}
}

func Test_AuthCustomHeader(t *testing.T) {
	// Sends authentication via custom header (X-API-Key)
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/auth_custom_header"
	result, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	if result.AuthHeaderSent != true {
		t.Errorf("equals mismatch: got %v", result.AuthHeaderSent)
	}
	if result.StatusCode != 200 {
		t.Errorf("equals mismatch: got %v", result.StatusCode)
	}
}
